workflows_core/engine/cluster_engine.py

```diff
@@ -46,15 +46,28 @@ def apply(self) -> Any:
         else:
             self._success_ratio = 1.0
 
-        # Update this in series
-        for i in range(self._num_chunks):
-            chunk = new_batch[i * self.pull_chunksize : (i + 1) * self._pull_chunksize]
-            self.update_chunk(
-                chunk,
+        batch_to_insert = []
+        payload_size = 0
+        upload_progress = 0
+
+        push_chunksize = get_optimal_chunksize(transformed_documents[:50])
```

Comment:
leave coment here please saying that we only do it for 50 because its an expensive operation